components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -88,7 +88,7 @@ void TrackingProtectionService::SetStartingSiteForRenderFrame(
 GURL TrackingProtectionService::GetStartingSiteForRenderFrame(
     int render_process_id,
     int render_frame_id) const {
-  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
```

Comment:
this changes things significantly because we can just access what we need directly on the UI thread. cc @jumde for a follow-up issue to simplify this code

Reply:
actually there might be a more immediate issue since https://github.com/brave/brave-core/pull/4374/files#diff-4318395fa4a47c9ce77b4ea0d452ffc1R104 is still on the IO thread.

Reply:
`render_frame_key_to_starting_site_url` is access on both the UI and IO thread which cannot happen

Reply:
I think we can move this code behind the buildflag `BRAVE_STP_ENABLED`